api/hasura/actions/create_circle.ts

```diff
@@ -0,0 +1,221 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { GraphQLClient, gql } from 'graphql-request';
+import Validator from 'validatorjs';
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
```

Comment:
We should strongly type these using Hasura's codegen features: https://hasura.io/docs/latest/graphql/core/actions/codegen/index.html

Reply:
Would i need to switch to node express framework to use hasura's codegen ?

Reply:
This PR has the relevant mutations w/ zeus generated code
https://github.com/coordinape/coordinape/pull/398

But I am splitting it up into a few PRs

Reply:
Hasura has a "typescript-zeit" generator that you can use for vercel style handler. Main thing to copy is the type definitions, ideally we can [create our own code generator](https://github.com/hasura/codegen-assets/blob/master/builder-kit/README.md) that does it in the exact right format we want.